test/pricer/pricer.test.ts

```diff
@@ -0,0 +1,174 @@
+import { expect } from 'chai';
+import {
+  restore,
+  stub,
+  SinonStubbedInstance,
+  createStubInstance,
+  SinonStub,
+} from 'sinon';
+import { BigNumber as BigNumberJs } from 'bignumber.js';
+import { getExchangeRate } from '../../src/pricer/pricer';
+import {
+  BaseExchangeApi,
+  CoinCodex,
+  CoinGecko,
+  RdocExchange,
+  TestExchange,
+} from '../../src/api';
+import * as pricerUtils from '../../src/pricer/utils';
+
+describe('pricer', function () {
+  describe('getExchangeRate', function () {
+    let coinGeckoStub: SinonStubbedInstance<CoinGecko>;
+    let fakeRifUsd: BigNumberJs;
+    let fakeRbtcUsd: BigNumberJs;
+    let fakeRbtcRif: BigNumberJs;
+    let fakeRifRbtc: BigNumberJs;
+    const RIF_SYMBOL = 'RIF';
+    const RBTC_SYMBOL = 'RBTC';
+    let createExchangeApiStub: SinonStub;
+
+    beforeEach(function () {
+      coinGeckoStub = createStubInstance(CoinGecko);
+      fakeRifUsd = randomBigNumberJs(25000);
+      fakeRbtcUsd = randomBigNumberJs(25000);
+      fakeRbtcRif = fakeRbtcUsd.dividedBy(fakeRifUsd);
+      fakeRifRbtc = fakeRifUsd.dividedBy(fakeRbtcUsd);
+      createExchangeApiStub = stub(pricerUtils, 'createExchangeApi');
+      createExchangeApiStub.withArgs('coinGecko').returns(coinGeckoStub);
+      stubExchangeRate(
+        coinGeckoStub,
+        RIF_SYMBOL,
+        pricerUtils.INTERMEDIATE_CURRENCY,
+        fakeRifUsd
+      );
+      stubExchangeRate(
+        coinGeckoStub,
+        RBTC_SYMBOL,
+        pricerUtils.INTERMEDIATE_CURRENCY,
+        fakeRbtcUsd
+      );
+    });
+
+    afterEach(function () {
+      restore();
+    });
+
+    function randomBigNumberJs(max: number) {
+      return BigNumberJs(Math.random() * max);
+    }
+
+    function stubExchangeRate(
+      exchangeApi: SinonStubbedInstance<BaseExchangeApi>,
+      sourceCurrency: string,
+      targetCurrency: string,
+      exchangeRate: BigNumberJs
+    ) {
+      exchangeApi.queryExchangeRate
+        .withArgs(sourceCurrency, targetCurrency)
+        .resolves(exchangeRate);
+    }
```

Comment:
I wouldn't do this. It doesn't do anything special and it just executes one fn with the args we're passing to it.

Reply:
updated